feat: pass X-Accel-* headers to client, fixes #6563

andriokha opened the pull request:
## The Issue

- #6563

By default, nginx [does not pass the header field `X-Accel-...` from the response of a FastCGI server to a client](http://nginx.org/en/docs/http/ngx_http_fastcgi_module.html#fastcgi_hide_header). It can be useful however during development for manual or automated testing of nginx-specific behavior. For example [Drupal's BigPipe test checks the value of the 'X-Accel-Buffering' header](https://git.drupalcode.org/project/drupal/-/blob/79a4904009909e83b94b6cbbfcc55600def67233/core/modules/big_pipe/tests/src/Functional/BigPipeTest.php#L346).

## How This PR Solves The Issue

It uses `fastcgi_pass_header` to forward the headers to the client.

## Manual Testing Instructions

Visit a page that sets one of the headers and verify you can see it in your client. [This gist tests all the headers except `X-Accel-Redirect`](https://gist.github.com/andriokha/38d92be900a3315861cc3fd8b8037ebd) (I don't know nginx well enough to coax it into giving a reply with that set.)

## Automated Testing Overview

No tests introduced, not sure if I should, don't knew DDEV well (:

## Release/Deployment Notes

Any `X-Accel-*` headers set by a PHP application will now be passed onto the client.

rfay reviewed (changes requested):
I'm pretty sure you'll want to do this in all the nginx configs in this directory. I think you're doing primarily drupal, so at least those.

rfay commented:
Please provide more specific manual testing instructions, like provide a gist or demo repo or something, thanks.

andriokha commented:
https://gist.github.com/andriokha/38d92be900a3315861cc3fd8b8037ebd

> I'm pretty sure you'll want to do this in all the nginx configs in this directory.

Ahh, I thought there was a base nginx config for all PHP types. I've again been bold and updated everything PHPey, happy to pull it back if desired.

> Please provide more specific manual testing instructions

I've added [a simple gist](https://gist.github.com/andriokha/38d92be900a3315861cc3fd8b8037ebd) that demonstrates the issue.

Thanks!

stasadev commented:
@andriokha, I pushed a new Docker image, but cannot update this PR, because access to `FabbDev/ddev` is locked.

Update this file on your end:

https://github.com/ddev/ddev/blob/fec8d695ffd89634060ed10d465ce6ba6f0f87ff/pkg/versionconstants/versionconstants.go#L14

```diff
-var WebTag = "20240918_saitho_imagemagic_remove_policy" // Note that this can be overridden by make
+var WebTag = "20240927_andriokha_nginx_headers" // Note that this can be overridden by make
```

And we can run the tests on this PR.

andriokha commented:
Rebased against `fec8d695ffd89634060ed10d465ce6ba6f0f87ff`.

andriokha commented:
I've run the test gist and that passed for all the affected types. I'll use it as my main driver today to smoke test. Thanks all!

rfay commented:
We can't rebase this, so please rebase. When you make a PR, please don't uncheck the default checkbox that allows other collaborators to push into your fork.

andriokha commented:
> please don't uncheck the default checkbox that allows other collaborators to push into your fork

I didn't.

stasadev commented:
That's because this PR is created from an org fork and there's nothing we can do about it:

- https://github.com/orgs/community/discussions/5634

rfay commented:
You did the right thing, thanks. But @stasadev why did we need a different `ddev-webserver` tag here? I only see changes in the embedded golang code.

stasadev commented:
> why did we need a different ddev-webserver tag here? I only see changes in the embedded golang code.

I saw the changed nginx config but didn't pay attention to where the changes were made.
Of course, it doesn't need a new tag, thanks for noticing this.

rfay commented on pkg/versionconstants/versionconstants.go:
```diff
@@ -11,7 +11,7 @@ var AmplitudeAPIKey = ""
 
 // WebTag defines the default web image tag
-var WebTag = "20241002_deviantintegral_vim_tiny" // Note that this can be overridden by make
+var WebTag = "20240927_andriokha_nginx_headers" // Note that this can be overridden by make
```
Sorry @andriokha this was a mistake to have required this. Now this will get in easier :)

```suggestion
var WebTag = "20241002_deviantintegral_vim_tiny" // Note that this can be overridden by make
```

rfay reviewed (approved):
Thanks!
